src/app/3d/qgs3dviewsmanagerdialog.cpp

```diff
@@ -0,0 +1,200 @@
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgs3dviewsmanagerdialog.h"
+
+#include "qgisapp.h"
+#include "qgs3dmapcanvasdockwidget.h"
+#include "qgsnewnamedialog.h"
+#include "qgs3dmapcanvas.h"
+#include "qgs3dviewsmanager.h"
+
+#include <QMessageBox>
+
+Qgs3DViewsManagerDialog::Qgs3DViewsManagerDialog( QWidget *parent, Qt::WindowFlags f )
+  : QDialog( parent, f )
+{
+  setupUi( this );
+
+  mListModel = new QStringListModel( this );
+  m3DViewsListView->setModel( mListModel );
+
+  m3DViewsListView->setEditTriggers( QAbstractItemView::NoEditTriggers );
+  m3DViewsListView->setSelectionMode( QAbstractItemView::SingleSelection );
+
+  connect( mButtonBox, &QDialogButtonBox::rejected, this, &QWidget::close );
+
+  connect( mShowButton, &QToolButton::clicked, this, &Qgs3DViewsManagerDialog::showClicked );
+  connect( mHideButton, &QToolButton::clicked, this, &Qgs3DViewsManagerDialog::hideClicked );
+  connect( mDuplicateButton, &QToolButton::clicked, this, &Qgs3DViewsManagerDialog::duplicateClicked );
+  connect( mRemoveButton, &QToolButton::clicked, this, &Qgs3DViewsManagerDialog::removeClicked );
+  connect( mRenameButton, &QToolButton::clicked, this, &Qgs3DViewsManagerDialog::renameClicked );
+  mShowButton->setEnabled( false );
+  mHideButton->setEnabled( false );
+
+  connect( m3DViewsListView->selectionModel(), &QItemSelectionModel::currentChanged, this, &Qgs3DViewsManagerDialog::currentChanged );
+
+  connect( QgsProject::instance()->views3DManager(), &Qgs3DViewsManager::viewsListChanged, this, &Qgs3DViewsManagerDialog::onViewsListChanged );
+  m3DViewsListView->selectionModel()->setCurrentIndex( m3DViewsListView->model()->index( 0, 0 ), QItemSelectionModel::Select );
+  currentChanged( m3DViewsListView->selectionModel()->currentIndex(), m3DViewsListView->selectionModel()->currentIndex() );
+}
+
+void Qgs3DViewsManagerDialog::onViewsListChanged()
+{
+  reload();
+}
+
+void Qgs3DViewsManagerDialog::showClicked()
+{
+  if ( m3DViewsListView->selectionModel()->selectedRows().isEmpty() )
+    return;
+
+  QString viewName = m3DViewsListView->selectionModel()->selectedRows().at( 0 ).data( Qt::DisplayRole ).toString();
+
+  Qgs3DMapCanvasDockWidget *widget = QgisApp::instance()->findChild<Qgs3DMapCanvasDockWidget *>( viewName + QStringLiteral( "DockObject" ) );
+
+  if ( !widget )
+  {
+    widget = QgisApp::instance()->open3DMapView( viewName );
+  }
+
+  if ( widget )
+  {
+    widget->show();
+    QgsProject::instance()->setDirty();
```

Comment:
```suggestion
```

Opening an existing view isn't really a change to the project -- I don't think it should cause a dirtying (no more so then opening/closing a layout designer window does)

Reply:
The 3D views are automatically opened at the start of the project unlike layour designers and showing a hidden project changes whether it opens on start or not. I think it should be dirtying the project

Reply:
I still disagree -- the purpose of this improvement is to separate the widget from the 3d map. So the widget state no longer becomes important anymore.

But happy for a second opinion here. Ping @wonder-sk @nirvn

Reply:
I tend to agree with @nyalldawson here -- we don't dirty the project when we pan / zoom the map canvas, it makes sense to not dirty the project when opening / closing a 3D map view now that the panel visibility is dissociated from the 3D map view's existence. Similarly, we wouldn't dirty up a project when we toggle an attribute table.

That being said, I can see how someone can strongly feel in the opposite opinion :) as you're pointing out @NEDJIMAbelgacem , 3D map views aren't exactly project layouts / attribute tables since their visibility state are restored upon project load.

My vote would be a +0.25 towards no dirtying to keep the dirtying concept tied to _data/settings modification_ vs. state change.

I'm not being very helpful now am I? :wink:

Reply:
One thing I want to add is that when the 3D view is closed, its settings are updated with the latest changes to the 3D Map view like the camera orientation and position. This is done so  that the next time the user opens the 3D map view will be back to the same configuration of when he closed it.

Reply:
@NEDJIMAbelgacem , well then it seems that would likely dirty the project, so you can remove setDirty here :)

Reply:
The is something else I want to say too: 
For now from the views manager standpoint the camera configuration and the rest of the configurations are not sperated and will need splitting the DOM into camera/non savable configuration and settings that are saved in the project.

@nirvn I'm a bit confused here, did you mean to keep or remove 😅 
I spoke with @wonder-sk and @uclaros and both of them seem to agree with my points - What do you think @nyalldawson ?

Reply:
Well, a view change shouldn't result in project dirtying either (i.e. zooming into a 2d map or moving the 3d map camera). The new position should definitely be saved if the user clicks to save the project, but a view change alone shouldn't classify as a modification to the project. (Just the same way as MS word won't mark a document as dirty if you open and scroll through a few pages, zoom in/out, etc). 

That's the way 2d canvas currently works, so 3d should also follow this approach